src/app/shared/form/builder/ds-dynamic-form-ui/models/checkbox/checkbox.model.ts

```diff
@@ -0,0 +1,27 @@
+import { DynamicFormControlLayout, serializable } from '@ng-dynamic-forms/core';
+import { DsDynamicInputModel, DsDynamicInputModelConfig } from '../ds-dynamic-input.model';
+
+export const DYNAMIC_FORM_CONTROL_TYPE_CUSTOM_CHECKBOX = 'CUSTOM_CHECKBOX';
+
+/**
+ * Main model for the 'CheckboxComponent' component.
+ * Only adding support for 'legend' and 'checked' properties.
+ * 
+ * @Author: Michaël Pourbaix <michael.pourbaix@uclouvain.be>
+ */
+
+export interface DynamicDsCheckboxModelConfig extends DsDynamicInputModelConfig {
+    legend: string;
+}
+
+export class DynamicDsCheckboxModel extends DsDynamicInputModel {
+    legend: string;
+    checked: boolean;
+    @serializable() readonly type: string = DYNAMIC_FORM_CONTROL_TYPE_CUSTOM_CHECKBOX;
```

Comment:
Maybe it could be of interest to create a variable `hideHint` (with a default value set to `false`) into `DsDynamicInputModel` and override this variable in this model.
Use this way, we could use `model.hideHint` instead of `model.type = 'CUSTOM_CHECKBOX'` into the `container.component.ts`